libs/features/mas/commerce/src/mas-commerce-service.js

```diff
@@ -0,0 +1,171 @@
+import { Checkout } from './checkout.js';
+import * as Constants from './constants.js';
+import { EVENT_TYPE_READY } from './constants.js';
+import { Defaults } from './defaults.js';
+import { Ims } from './ims.js';
+import { getPriceLiterals } from './literals.js';
+import { Log } from './log.js';
+import { Price } from './price.js';
+import { getSettings } from './settings.js';
+import { Wcs } from './wcs.js';
+import { setImmediate } from './utilities.js';
+
+export const TAG_NAME_SERVICE = 'mas-commerce-service';
+
+/**
+ * Custom web component to provide active instance of commerce service
+ * to consumers, appended to the head section of current document.
+ */
+export class MasCommerceService extends HTMLElement {
+    static instance;
+    promise = null;
+
+    get config() {
+        const { searchParams } = new URL(import.meta.url);
+        const env = this.getAttribute('env') || searchParams.get('env');
+        const isStage = env?.toLowerCase() === 'stage';
+        const envName = isStage ? 'stage' : 'prod';
+        const commerceEnv = isStage ? 'STAGE' : 'PROD';
+        const config = {
+            env: { name: envName },
+            commerce: { 'commerce.env': commerceEnv },
+        };
+        //root parameters
+        ['locale', 'country', 'language'].forEach((attribute) => {
+            const value = this.getAttribute(attribute);
+            if (value) {
+                config[attribute] = value;
+            }
+        });
+        //commerce parameters
+        ['checkoutWorkflowStep', 'forceTaxExclusive', 'checkoutClientId'].forEach((attribute) => {
+            const value = this.getAttribute(attribute);
```

Comment:
I think we should adopt html friendly attribute names: `checkout-workflow-step`, `force-tax-exclusive` `checkout-client-id`